src/main/kotlin/no/nav/amt/arena/acl/processors/HistDeltakerProcessor.kt

```diff
@@ -112,18 +142,61 @@ open class HistDeltakerProcessor(
 			operation = AmtOperation.MODIFIED,
 			payload = deltaker
 		)
-
+		deltaker.validerGyldigHistDeltaker()
 		kafkaProducerService.sendTilAmtTiltak(deltaker.id, deltakerKafkaMessage)
 		log.info("Melding for hist-deltaker id=${deltaker.id} arenaHistId=${arenaDeltaker.tiltakdeltakerId} transactionId=${deltakerKafkaMessage.transactionId} op=${deltakerKafkaMessage.operation} er sendt")
 	}
 
-	private fun getAmtDeltaker(
-		personIdent: String,
-		gjennomforingId: UUID,
-		startdato: LocalDate?,
-		sluttdato: LocalDate?
-	): DeltakerDto? {
+	private fun getAmtDeltaker(id: UUID, personIdent: String): DeltakerDto? {
 		val deltakelser = amtTiltakClient.hentDeltakelserForPerson(personIdent)
-		return deltakelser.find { it.gjennomforing.id == gjennomforingId && it.startDato == startdato && it.sluttDato == sluttdato }
+		return deltakelser
+			.find { it.id == id }
 	}
+
+	private fun getMatchingDeltaker(
+		arenaHistDeltaker: ArenaHistDeltaker,
+	): DeltakerDbo? {
+		val personId = arenaHistDeltaker.PERSON_ID
+			?: throw ValidationException("Kan ikke matche hist deltaker som mangler PERSON_ID")
+		val modDato = arenaHistDeltaker.MOD_DATO?.asLocalDateTime()
+			?: throw ValidationException("Kan ikke matche hist deltaker som mangler MOD_DATO")
+		val datoFra = arenaHistDeltaker.DATO_FRA?.asLocalDate()
+		val datoTil = arenaHistDeltaker.DATO_TIL?.asLocalDate()
+		val arenaDeltakere = deltakerRepository
+			.getDeltakereForPerson(personId, arenaHistDeltaker.TILTAKGJENNOMFORING_ID)
+
+		val matchendeDeltakere = arenaDeltakere
+			.filter { it.datoFra == datoFra }
+			.filter { it.datoTil == datoTil }
+			.filter { it.modDato == modDato }
```

Comment:
Bør vi prøve å runde av mod-datoen slik at vi f.eks. ikke har med sekunder eller hvertfall millisekunder? Synes ofte jeg opplever at to localdatetimes som egentlig er like ikke er nøyaktig like lenger hvis de blir skrevet til databasen og lest ut igjen etterpå..

Reply:
Ja, blir mye forvirring med det her og er spent på om det funker. Ser at mod dato og reg dato mangler millisekunder sånn at jeg tror det i praksis kan gå fint, ser at de blir lagret sånn her i arena  2024-06-26 12:37:35 og vår db: 2021-04-13 00:00:24.000000 +00:00 når vi bruker .asLocalDateTime.
Kan kanskje hende det blir krøll når vi leser ut fra databasen? Sjekker litt
Er ikke sikker på om jeg tør å fjerne sekundene fra sammenlikningen.

Reply:
Dette kommer ikke til å fungere likevel fordi MOD_DATO i hist er ofte lik som REG_DATO, så MOD_DATO blir nok satt når deltakeren historiseres. Så da kommer Plan B matche på dato_statusendring og fra og tildato.
Her finner jeg ingen match på noen i både hist_tiltakdeltaker og tiltakdeltaker i q0(som ventet)
Og jeg ser at dato_statusendring er FØR MOD_DATO